### ISSUE TITLE
[Browser Plugin Connector] Implement Sign request Popup in Status desktop

### ISSUE THREAD

**Issue by kounkou** [status-im/status-desktop#15172]
### Description

Part of Epic https://github.com/status-im/status-desktop/issues/14794

Implement the sign request popup in status desktop that aims at signing requests emitted by the user in the browser plugin.

### Acceptance criteria

- Popup matches the Figma

### Figma

![Screenshot from 2024-06-13 13-36-00](https://github.com/status-im/status-desktop/assets/2589171/e927b997-be70-49a5-b5c8-623529f60af5)

![Screenshot from 2024-06-13 13-41-01](https://github.com/status-im/status-desktop/assets/2589171/ef45e2b4-efad-4bfc-97cd-f429174952f6)

https://www.figma.com/design/1OYKMzU6KTQHQAqDhojk0r/Status-connector?node-id=2015-28681&t=j9KXTb1Y6jPZPh1u-0

https://www.figma.com/design/1OYKMzU6KTQHQAqDhojk0r/Status-connector?node-id=2015-30099&t=Im8bj2UcUAPndqnI-0

**Comment by caybro** [comment]
Are you guys in sync @noeliaSD ? :) I'm having a dejavu :D

**Comment by jrainville** [comment]
@kounkou I guess these issues are for the UI part (ie the UI team)? Since you specify the figma.

**Comment by noeliaSD** [comment]
Hey hey team!! @jrainville @kounkou I think we need to sync here!! We have this [epic](https://github.com/status-im/status-desktop/issues/14588) already ongoing for `dApps` and maybe we are duplicating things!! Please, take a look at it so that you know what will be already covered by it!

For example, this task: https://github.com/status-im/status-desktop/issues/14762 is similar than the one described in here, but here we have some specific particularities for your topic. Maybe we can link them both and make this task blocked by #14762. Once we are done with it, the basic dialog with the same structure will be ready to be reused in here. Then the only effort for this task will be just adding the specific details for your use case. WDYT?

**Comment by noeliaSD** [comment]
> @kounkou I guess these issues are for the UI part (ie the UI team)? Since you specify the figma.

@jrainville, if this is a use case to cover by this milestone, the `ui-team` doesn't have capacity for it... :(

**Comment by noeliaSD** [comment]
Depends on: https://github.com/status-im/status-desktop/issues/14762

**Comment by jrainville** [comment]
> Hey hey team!! @jrainville @kounkou I think we need to sync here!! We have this [epic](https://github.com/status-im/status-desktop/issues/14588) already ongoing for `dApps` and maybe we are duplicating things!! Please, take a look at it so that you know what will be already covered by it!
> 
> For example, this task: #14762 is similar than the one described in here, but here we have some specific particularities for your topic. Maybe we can link them both and make this task blocked by #14762. Once we are done with it, the basic dialog with the same structure will be ready to be reused in here. Then the only effort for this task will be just adding the specific details for your use case. WDYT?

I agree. Let's just keep it as blocked for now since you'll already do the basic modal for it.

> if this is a use case to cover by this milestone, the ui-team doesn't have capacity for it... :(

Since you'll have the basic dialog, we can do the finishing touches of integration for this.

Thanks

**Comment by stefandunca** [comment]
Note that the basic implementation for this control is the [ConnectDAppModal.qml](https://github.com/status-im/status-desktop/blob/master/ui/imports/shared/popups/walletconnect/ConnectDAppModal.qml) which I'm now extending for supporting sign and send transaction.

The one we implemented is the descoped one and the ones you have in the screenshot from figma seems like the original non-descoped version. We dropped the ABI and HEX in favor of showing the raw formated json of the request to sign/validate.
